app/assets/javascripts/app_router.js

```diff
@@ -1,22 +1,48 @@
 var Pokedex = require("./models/pokedex.js");
 var PokedexView = require("./pokedex_view.js");
+var PokemonShowView = require("./pokemon_show_view.js");
 
-var AppRouter = function ($element) {
+var AppRouter = function (routes) {
   this.pokedex = new Pokedex();
-  this.pokedex.fetchThemAll(this.renderPokedex.bind(this));
-	this.$el = $element;
-  this.renderPokedex();
+  this.routes = routes;
+  this.container;
+};
+
+AppRouter.prototype.go = function (route) {
+  var regexRoute = /[\w-]+\//;
+  var routeName = route.match(regexRoute)[0];
+  var regexId = /\/(\d+)/;
+  var id = route.match(regexId) == null ? null : parseInt(route.match(regexId)[1]);
```

Comment:
A comment for later: these REGEXes are probably going to be very confusing for our students. We can use something like [routington](https://www.npmjs.com/package/routington) or [pathparser](https://www.npmjs.com/package/pathparser) to abstract that away.